modules/xray-udp-exporter/src/main/scala/io/janstenpickle/trace4cats/xray/XRayUdpSpan.scala

```diff
@@ -0,0 +1,167 @@
+package io.janstenpickle.trace4cats.xray
+
+import cats.{Applicative, Eval, Functor, Monad}
+import cats.data.NonEmptyList
+import cats.effect.{Clock, Ref}
+import cats.effect.std.Random
+import cats.syntax.applicative._
+import cats.syntax.apply._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.show._
+import cats.syntax.traverse._
+import fs2.Chunk
+import io.circe.{Encoder, Json, JsonObject, Printer}
+import io.circe.syntax._
+import io.janstenpickle.trace4cats.`export`.SemanticTags
+import io.janstenpickle.trace4cats.model._
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import scala.util.matching.Regex
+
+private[xray] object XRayUdpSpan {
+
+  /** Span names can contain Unicode letters, numbers, and whitespace, and a limited set of symbols.
+    *
+    * See the `name` documentation under "Required Segment Fields"
+    * https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-fields
+    */
+  private val nameRegex: Regex = """[^\p{L}0-9 _\.:/%&#=\+\-@]""".r
```

Comment:
This and the next val's names are a bit misleading. Could we name them differently? My suggestion is a bit verbose but is more clear. I also fixed escaping, it seems to have been wrong wrt to `\` and `.`.
```suggestion
  private val nameIllegalCharRegex: Regex = """[^\p{L}0-9 _.:/%&#=\\+\-@]""".r
```

Reply:
I think `nameIllegalCharRegex` actually sounds like the opposite of what this is doing -- the `^` directly after `[` indicates that the regex will match any characters *except* the ones listed here. How does `nameAllowedCharRegex` sound instead?

Re: escaping, I wasn't trying to include `\` in the list of characters, I was just escaping `.` and `+`, but I think it's unnecessary within `[]`, so I can remove them. My suggestion is:

```suggestion
  private val nameAllowedCharRegex: Regex = """[^\p{L}0-9 _.:/%&#=+\-@]""".r
```

Reply:
Yes, this regex matches any character except for those listed in `[ ]`, i.e. _special_ characters that _cannot_ be part of a name. Hence it's a list of illegal characters. Right?

Reply:
Re: escaping

> name – The logical name of the service that handled the request, up to 200 characters. For example, your application's name or domain name. Names can contain Unicode letters, numbers, and whitespace, and the following symbols: _, ., :, /, %, &, #, =, +, \\, -, @

As you see `\` should be in the list.

Reply:
Thanks, I had missed that. And I don't feel strongly about the names, so updated to use `*IllegalChar*`